inc/api.class.php

```diff
@@ -658,99 +694,154 @@ protected function getItem($itemtype, $id, $params = []) {
+                  'LEFT JOIN' => [
+                     "$netport_table AS netp_subtable" => [
+                        'ON' => [
+                           'netp_subtable'   => 'networkports_id',
+                           'netp'            => 'id'
+                        ]
+                     ]
+                  ],
+                  'WHERE'     => [
+                     'netp.instantiation_type'  => $networkport_type,
+                     'netp.items_id'            => $id,
+                     'netp.itemtype'            => $itemtype,
+                     'netp.is_deleted'          => 0
+                  ]
+               ]);
+
+               while ($data = $netp_iterator->next()) {
+                  if (isset($data['netport_id'])) {
+                     // append network name
+                     $concat_expr = new QueryExpression(
+                        "GROUP_CONCAT(CONCAT(ipadr.`id`, '".Search::SHORTSEP."' , ipadr.`name`)
+                        SEPARATOR '".Search::LONGSEP."') AS ipadresses"
+                     );
+                     $netn_iterator = $DB->request([
+                        'SELECT'    => [
+                           $concat_expr,
+                           'netn.id AS networknames_id',
+                           'netn.name AS networkname',
+                           'netn.fqdns_id',
+                           'fqdn.name AS fqdn_name',
+                           'fqdn.fqdn'
+                        ],
+                        'FROM'      => [
+                           'glpi_networknames AS netn'
+                        ],
+                        'LEFT JOIN' => [
+                           'glpi_ipaddresses AS ipadr'               => [
+                              'ON' => [
+                                 'ipadr'  => 'items_id',
+                                 'netn'   => 'id'
+                              ]
+                           ],
+                           'glpi_fqdns AS fqdn'                      => [
+                              'ON' => [
+                                 'fqdn'   => 'id',
+                                 'netn'   => 'fqdns_id'
+                              ]
+                           ],
+                           'glpi_ipaddresses_ipnetworks AS ipadnet'  => [
+                              'ON' => [
+                                 'ipadnet'   => 'ipaddresses_id',
+                                 'ipadr'     => 'id'
+                              ]
+                           ],
+                           'glpi_ipnetworks AS ipnet'                => [
+                              'ON' => [
+                                 'ipnet'     => 'id',
+                                 'ipadnet'   => 'ipnetworks_id'
+                              ]
+                           ]
+                        ],
+                        'WHERE'     => [
+                           'ipadr.itemtype'  => 'NetworkName',
```

Comment:
Why moving this condition here instead of keeping it on the join clause ?

Reply:
Iterator does not support that.

Reply:
It does, see https://github.com/glpi-project/glpi/blob/9.3/bugfixes/tests/units/DBmysqlIterator.php#L367

Reply:
Indeed, I forget this has been added :D

Anyways, this does not change anything to the queries themselves, and since I'm already on some other conversions, I prefer let this one as is ;)

Reply:
In fact, it does have an impact if condition applies to a `LEFT JOIN`.
See https://www.db-fiddle.com/f/7XJVuPnUKXgufZngbLirJ7/1 .